experimental/uefi/baremetal/src/hvm_start_info.rs

```diff
@@ -0,0 +1,79 @@
+//
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+/* Generated with:
+ * $ bindgen --no-layout-tests --use-core \
+ *   --ctypes-prefix c_types include/public/arch-x86/hvm/start_info.h
```

Comment:
Would it work / make sense to actually do this in a build.rs file? Something like https://github.com/project-oak/oak/blob/fbb62f1303481506474a899ba0bc3d45e2da6a29/oak_idl_gen_services/src/lib.rs#L76

Reply:
Maybe, I'll have to read up on bindgen a bit (you *can* call it from `build.rs`, that's the recommended way to do it).

However, that implies:
a) we will bundle the C header files from Xen and Linux with our code;
b) the files have some manual modifications (such as implementing the traits and overriding `c_types`), not sure how that works with bindgen.

For now, I'll leave it be, but yes, generating it from authoritative source would be a good idea.

Reply:
I'm now generating it using bindgen during build time.